File: Atividades/Lista2/C++/conversaoDeTemperatura.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <iomanip>

using namespace std;

class Temperatura {
private:
    double temperatura;

public:
    Temperatura(double temperatura) {
        this->temperatura = temperatura;
    }

    string toString() const {
        ostringstream stream;
        stream << fixed << setprecision(1) << temperatura;
        return stream.str();
    }

    void toFahrenheit() {
        temperatura = temperatura * 9.0/5.0 + 32;
    }

    void toCelsius() {
        temperatura = (temperatura - 32) * 5.0/9.0;
    }
};

int main() {
    double temp;
    cin >> temp;

    Temperatura temperatura(temp);
    
    temperatura.toFahrenheit();
    cout << "temperatura: " << temperatura.toString() << " graus fahrenheit" << endl;

    temperatura.toCelsius();

    cout << "temperatura: " << temperatura.toString() << " graus celsius\n";

    return 0;
}

File: Atividades/Lista_2/Java/ConversaoDeTemperatura.java
package Atividades.Lista_2.Java;

import java.util.Scanner;

public class ConversaoDeTemperatura {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        double temp = input.nextInt();
        input.nextLine();

        Temperatura temperatura = new Temperatura(temp);

        temperatura.toFahrenheit();

        System.out.printf("temperatura: %s graus fahrenheit\n", temperatura.toString());

        temperatura.toCelsius();

        System.out.printf("temperatura: %s graus celsius\n", temperatura.toString());

        input.close();
    }
}

class Temperatura {
    private double temperatura;

    public Temperatura(double temperatura) {
        this.temperatura = temperatura;
    }

    public String toString() {
        return "" + temperatura;
    }

    public void toFahrenheit() {
        temperatura = temperatura * 9.0/5.0 + 32;
    }

    public void toCelsius() {
        temperatura = (temperatura - 32) * 5.0/9.0;
    }
}